File: 1974-minimum-time-to-type-word-using-special-typewriter/1974-minimum-time-to-type-word-using-special-typewriter.java
class Solution {
    public int minTimeToType(String word) {
        char[] chars = word.toCharArray();
        char current = 'a';
        int cost = 0;
        for (char c : chars) {
            if (c == current) {
                cost++;
            } else {
                int diff = Math.abs(c - current);
                if (diff > 12) cost += (26 - diff);
                else cost += diff;
                cost++;
                current = c;
            }
        }
        return cost;
    }
}

File: 1974-minimum-time-to-type-word-using-special-typewriter/1974-minimum-time-to-type-word-using-special-typewriter.kt
import java.lang.Math.min
import kotlin.math.abs

class Solution {
    fun minTimeToType(word: String): Int {

        var answer: Int = 0
        var prev:Char = 'a'

        for (wordChar in word.toCharArray()) {
            val forward = abs(wordChar - prev)
            val backward = abs(26 - forward)
            answer += forward.coerceAtMost(backward) + 1
            prev = wordChar
        }
        
        return answer

    }
}
